src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -44,12 +44,12 @@ internal bool IsUnprocessedBufferEmpty()
             return _startIndex >= _endIndex; // Everything has been processed;
         }
 
-        internal unsafe void AppendExtraBuffer(byte* buffer, int bufferLength)
+        internal void AppendExtraBuffer(ReadOnlySpan<byte> buffer)
         {
             // Then convert the bytes to chars
-            int charLen = _encoding.GetMaxCharCount(bufferLength);
-            char* charPtr = stackalloc char[charLen];
-            charLen = _encoding.GetChars(buffer, bufferLength, charPtr, charLen);
+            Span<char> chars = stackalloc char[_encoding.GetMaxCharCount(buffer.Length)];
```

Comment:
Unrelated to your changes, but shouldn't this be guarding against large char counts (perhaps using an assert)?

Reply:
Possibly?  I'm not entirely sure what's preventing large inputs; from a quick glance, it seems like the max buffer that's passed in comes from a pre-allocated array of a fixed max size.

Reply:
It's probably controlled by the [TransferBytes](https://github.com/dotnet/runtime/pull/42714/files/959432be0d0019d59834beff10eec475b925f381#diff-b245310c30b4892f1e122bbad2a227b3L661) method which copied the buffer one byte at a time.

Reply:
It's also used here:
https://github.com/dotnet/runtime/blob/e17d818e25567d7fa9525ae5d32e2b8d2483ae96/src/libraries/System.Console/src/System/IO/StdInReader.cs#L446
so a limit of 1024.

Reply:
Perhaps an assert is appropriate then